File: thinking_and_programming_in_functional_style/code/clojure/exercise2/exercise.clj
;Given a list of names, find the total number of characters. For example, given “John”, “Jack”, “Jill”, “Sam”, “William”, the result should be 22.

(def names (list "John" "Jack" "Jill" "Sam" "William"))

(println (reduce + (map count names)))

;or

(println (count (apply str names)))

File: thinking_and_programming_in_functional_style/code/scala/exercise1/exercise.scala
//Given a list of names, produce a list of tuples with names and size of each name.
//For example, given “John”, “Jack”, “Jill”, “Sam”, “William”, the result should be
//the list (“John”, 4), (“Jack, 4), (“Jill”, 4), (“Sam”, 3), (“William, 7).

val names = List("John", "Jack", "Jill", "Sam", "William")

println(names.map { e => (e, e.length()) })